core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -367,9 +367,15 @@ private[spark] object JsonProtocol {
         ("Job ID" -> taskCommitDenied.jobID) ~
         ("Partition ID" -> taskCommitDenied.partitionID) ~
         ("Attempt Number" -> taskCommitDenied.attemptNumber)
-      case ExecutorLostFailure(executorId, isNormalExit) =>
-        ("Executor ID" -> executorId) ~
-        ("Normal Exit" -> isNormalExit)
+      case ExecutorLostFailure(executorId, isNormalExit, reason) =>
+        if (reason.isDefined) {
+          ("Executor ID" -> executorId) ~
+          ("Normal Exit" -> isNormalExit) ~
+          ("Loss Reason" -> reason.get.toString)
```

Comment:
Actually, while going through the example [here](https://github.com/json4s/json4s#example), I decided to use this simpler looking snippet:
````      
      case ExecutorLostFailure(executorId, isNormalExit, reason) =>
        ("Executor ID" -> executorId) ~
        ("Normal Exit" -> isNormalExit) ~
        ("Loss Reason" -> reason.map(_.toString))
````